### ISSUE TITLE
gem not compiling on Alpine Linux 3.4

### ISSUE THREAD

**Issue by danieloliveira079** [scoutapp/scout_apm_ruby#125]
I am building a docker rails image based on ruby:2.4-alpine3.4.

When running `bundle install --without development test` the following it output. I could not find any reference that gives me a clue about what I am missing. Any help is really appreciated.

Apart than the default packages provided within the base image I am also installing:

`RUN apk add --no-cache git libxml2-dev postgresql-dev build-base `

output:
```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /usr/local/bundle/gems/scout_apm-3.0.0.pre9/ext/stacks
/usr/local/bin/ruby -r ./siteconf20170920-20724-1i4x96b.rb extconf.rb
checking for rb_postponed_job_register_one()... yes
checking for rb_profile_frames()... yes
checking for rb_profile_frame_absolute_path()... yes
checking for rb_profile_frame_label()... yes
checking for rb_profile_frame_classpath()... yes
checking for __linux__... yes
checking for main() in -lrt... yes
checking for stdatomic.h... yes
creating Makefile

current directory: /usr/local/bundle/gems/scout_apm-3.0.0.pre9/ext/stacks
make "DESTDIR=" clean

current directory: /usr/local/bundle/gems/scout_apm-3.0.0.pre9/ext/stacks
make "DESTDIR="
compiling stacks.c
stacks.c: In function 'init_thread_vars':
stacks.c:303:23: error: 'SIGEV_THREAD_ID' undeclared (first use in this function)
   _sev.sigev_notify = SIGEV_THREAD_ID;
                       ^
stacks.c:303:23: note: each undeclared identifier is reported only once for each function it appears in
stacks.c:305:7: error: 'struct sigevent' has no member named '_sigev_un'
   _sev.sigev_notify_thread_id = syscall(SYS_gettid);
       ^
Makefile:238: recipe for target 'stacks.o' failed
make: *** [stacks.o] Error 1

make failed, exit code 2

Gem files will remain installed in /usr/local/bundle/gems/scout_apm-3.0.0.pre9 for inspection.
Results logged to /usr/local/bundle/extensions/x86_64-linux/2.3.0/scout_apm-3.0.0.pre9/gem_make.out

An error occurred while installing scout_apm (3.0.0.pre9), and Bundler cannot continue.
Make sure that `gem install scout_apm -v '3.0.0.pre9'` succeeds before bundling.

In Gemfile:
  scout_apm
```

**Comment by cschneid** [comment]
We've run across this (and is one of the big reasons that 3.0 is still in beta). It's due to Alpine linux using a different libc which doesn't support the threading C code we are using.  It's likely possible to fix, but we haven't had a chance to dig into it.

For now, either use the stable scout_apm 2.1.32, or build off an Ubuntu based docker image.

**Comment by danieloliveira079** [comment]
This is exactly the point @cschneid. My attempt here is to make an smaller rails image.

And as far as I could notice there are many more gems having the same issue.

Any suggestion in this direction would be appreciated. Rails images based on debian or ubuntu becomes huge in size. You can imagine the drawbacks.

**Comment by danieloliveira079** [comment]
Downgraded to 2.1.32 solved the problem. Thank you I will close the issue.

**Comment by ankitsinghaniyaz** [comment]
I have the same issue on Ubuntu 18.04

**Comment by cschneid** [comment]
Hi @ankitsinghaniyaz can you create a new issue on this repo with the output of the failed compilation?

The underlying issue with Alpine linux is that they use a different libc implementation from Ubuntu - whatever's going on in Ubuntu is likely something different.
